jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -438,6 +442,49 @@ public CRIUSupport registerRestoreEnvFile(Path envFile) {
 		return this;
 	}
 
+	/**
+	 * Add new JVM options upon restore. The options will be specified in an options
+	 * file with the form specified in https://www.eclipse.org/openj9/docs/xoptionsfile/
+	 *
+	 * TODO: The first iteration will only support -D options. Subsequent iterations will add more functionality.
+	 *
+	 * @param optionsFile The file that contains the new JVM options to be added on restore
+	 *
+	 * @return this
+	 */
+	public CRIUSupport registerRestoreOptionsFile(Path optionsFile) {
+		String optionsFilePath = optionsFile.toAbsolutePath().toString();
+		this.optionsFile = "-Xoptionsfile=" + optionsFilePath; //$NON-NLS-1$
+
+		J9InternalCheckpointHookAPI.registerPostRestoreHook(RESTORE_OPTIONS_FILE_PRIORITY,
+				"Restore options via options file:" + optionsFilePath, () -> { //$NON-NLS-1$
+					try (BufferedReader optionsFileReader = new BufferedReader(new FileReader(optionsFilePath))) {
+						String entry = null;
+						while ((entry = optionsFileReader.readLine()) != null) {
+							entry = entry.trim();
+							if (!entry.isEmpty()) {
+								if (entry.startsWith("-D")) { //$NON-NLS-1$
+									String entrySplit[] = entry.split("=", 2); //$NON-NLS-1$
+									String propertyValue = entrySplit[1];
```

Comment:
This will throw an exception if the line doesn't contain "="; probably not the best user experience.
Perhaps adding to the catch list is reasonable, but it would be helpful to include some of the troublesome input in the exception thrown there.